pkg/controllers/workgenerator/controller_integration_test.go

```diff
@@ -1636,7 +1650,440 @@ var _ = Describe("Test Work Generator Controller", func() {
 		})
 	})
 
-	// TODO: add a test for the apply strategy
+	Context("Sync apply strategy changes", Ordered, func() {
+		var clusterResourceBinding *placementv1beta1.ClusterResourceBinding
+
+		BeforeAll(func() {
+			memberClusterName = "cluster-" + utils.RandStr()
+			memberCluster := clusterv1beta1.MemberCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: memberClusterName,
+				},
+			}
+			Expect(k8sClient.Create(ctx, &memberCluster)).Should(Succeed(), "Failed to create member cluster")
+
+			memberClusterNamespaceName = fmt.Sprintf(utils.NamespaceNameFormat, memberClusterName)
+			memberClusterReservedNS := corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: memberClusterNamespaceName,
+				},
+			}
+			Expect(k8sClient.Create(ctx, &memberClusterReservedNS)).Should(Succeed(), "Failed to create namespace")
+
+			testCRPName = "crp-" + utils.RandStr()
+		})
+
+		It("can create master resource snapshot and resource binding", func() {
+			// Create the master resource snapshot.
+			masterResourceSnapshot := generateResourceSnapshot(
+				1, 1, 0,
+				[][]byte{
+					testResourceCRD, testNameSpace, testResource,
+				})
+			Expect(k8sClient.Create(ctx, masterResourceSnapshot)).Should(Succeed(), "Failed to create master resource snapshot")
+
+			resourceBindingSpec := placementv1beta1.ResourceBindingSpec{
+				State:                placementv1beta1.BindingStateBound,
+				ResourceSnapshotName: masterResourceSnapshot.Name,
+				TargetCluster:        memberClusterName,
+			}
+			createClusterResourceBinding(&clusterResourceBinding, resourceBindingSpec)
+		})
+
+		It("should create work object with expected apply strategy", func() {
```

Comment:
```suggestion
		It("should create work object with no apply strategy", func() {
```

Reply:
when does the binding have no applyStrategy?

Reply:
Hi Ryan! `ApplyStrategy` is a optional pointer field, so a `nil` apply strategy can be quite common.

Reply:
Edited the It spec name as suggested.